### DISCUSSION TITLE
let onyxia administrator modify values.schema.json without recreate a chart

### DISCUSSION THREAD

**Discussion by fcomte** [InseeFrLab/legacy-onyxia-entrypoint#62] [Q&A] [answered]
For an administrator of Onyxia, reducing options offer to users is very complexe
They need to do their own charts even if there is no modification to the chart. 
If an administrator want less or more options for a community chart it's very uncomfortable to create a whole chart for that.

Maybe onyxia-api can offer the options to administrator another source of truth for values.schema.json
This is clearly touchy because values.schema.json must be coherent with the version of a chart.. so surcharging it may be very complexe and boring for an administrator when the chart evolved. 

worth / not worth? what do you think ?

**Comment by gmougeolle** [comment, marked as answer]
I believe our project is somewhat involved in this question. In our case, it's much easier to simply hide the values in the front-end (with an alternative UI in our case) than to maintain a simpler `values.schema.json`. If any other Onyxia has this need, it would be interesting to see which options they want to hide, and maybe then decide of a way in the JSON schema to allow for deciding the complexity of each configurable option, and then simply allow certain users / regions with an "advanced" mode that shows more options ?
Maybe some sort of levels akin to debugging values (0 - very verbose, 5 - only essentials) ?

**Reply by fcomte** [reply]
i think it's a better idea than maintaining an isolated values.schema.json, thx !

**Comment by alexisdondon** [comment]
Is it a worthy feature for now, compare to a viable alternative (not sur):
- admin make his own catalog with same folder with only a chart dependency fixed version to the sspcloud one.
- make his restricted values.schema.json besides this dependency helm chart.

While the dependency is fixed no problem of compatibility until the admin decide to upgrade the pinned version.

**Reply by fcomte** [reply]
I don't really know , I don't like to have another level for the all the values : mainchart.resources...
